### PR TITLE
Pdbecif correction

### PR DESCRIPTION
**Pull request by roshkjr**
Solves issue [10](https://github.com/PDBeurope/ccdutils/issues/10)

### CODE REVIEW

File: pdbeccdutils/core/ccd_writer.py [hunk 1]
```diff
@@ -1188,27 +1186,31 @@ def _add_fragments_and_scaffolds_cif(component, cif_block_copy):
...
+        inchikey = rdkit.Chem.MolToInchiKey(mol)
         new_row = [
             component.id,
             scaffold.name,
             f"S{i+1}",
             "scaffold",
             scaffold.smiles,
-            rdkit.Chem.MolToInchi(mol),
-            rdkit.Chem.MolToInchiKey(mol),
+            inchi if inchi else None,
```
Comment (otienoanyango) [inline pdbeccdutils/core/ccd_writer.py:1197]: You can also simplify it as just `inchi or None` but this is okay

File: pdbeccdutils/tests/test_file_writing.py [hunk 2]
```diff
@@ -209,3 +211,43 @@ def test_plain_cif_write(component: Component, tmpdir, rem_hs):
...
+            cif_block = cif.read(str(path)).sole_block()
+            assert cif_block
+            assert component.id == cif_block.name
+            cif_categories = cif_block.get_mmcif_category_names()
+            if component.scaffolds:
+                assert "_pdbe_chem_comp_substructure." in cif_categories
+                substructure = cif_block.get_mmcif_category(
+                    "_pdbe_chem_comp_substructure."
+                )
+                for _, values in substructure.items():
```
Comment (otienoanyango) [inline pdbeccdutils/tests/test_file_writing.py:248]: Look at Oliver's test in the other pull request and may be think of how to incorporate it. It seems to test "''" which is a good test here.

**Review by otienoanyango — APPROVED** [review]
In future, it may be better to adjust someone's pull request rather than create a separate one so their effort is honored

**Comment by osmart** [comment]
Hi Roshan,

Thanks for your reply https://github.com/PDBeurope/ccdutils/commit/6fa779f8d6564a65fa41471168ef45570a3dff2b#commitcomment-108723106

Here are some tips from an 'old-time' programmer:

**Comment by osmart** [comment]
## Keep commits atomic and make the 1st line of your commit messages human-readable.

Change one thing at a time. Your message for this commit starts "Pdbecif #12" this is meaningless to humans - the commit message says what the commit actually does in human terms. It is great to include the issue number as well perhaps further down.  You commit message includes a list of 5 things that are done. I would suggest it is better to do 5 commits.

There are a number of reasons why this is a good idea:
* if you ever need to revert a change (either permanently or for a test) this can be easily achieved. If you commit mulitple things in the same commit then this becomes difficult.
* git blame is much more useful

Remember that code is "code is written once but read many times". Using git commits carefully helps this process.

**Comment by osmart** [comment]
## Read (or review) "the Zen of Python" - it has some useful pointers for this bug and fix

If you are not familiar with it the Zen of Python can be got at every Python prompt with `import this` or 

https://docs.python-guide.org/writing/style/#zen-of-python

For this bug fix I think that there are a couple of points that are relevant...

**Comment by osmart** [comment]
In `ccd_writer.py` I proposed:
https://github.com/PDBeurope/ccdutils/blob/e2e795ab6a1645452b9c40df01496fd4f17f8aa4/pdbeccdutils/core/ccd_writer.py#L1203-L1219

But you implemented: https://github.com/PDBeurope/ccdutils/blob/6fa779f8d6564a65fa41471168ef45570a3dff2b/pdbeccdutils/core/ccd_writer.py#L1187-L1199

I would contend that **"Errors should never pass silently."** (from Zen of Python). Unfortunately in the RDKit API a choice was made for `Chem.MolToInch` to by default return an empty string if an error was found in working out the InChI. This was a pretty bad choice, particularly for the InChiKey. By using `treatWarningAsError=True` I think that the code is much more readable and therefore easier to maintain in the future.

Some programmers take pride in writing code that is short as possible

**Comment by osmart** [comment]
## Testing

Stephen commented about this above https://github.com/PDBeurope/ccdutils/pull/12#discussion_r1163668455

The suggested test `test_hem_cif_for_invalid_cif_missing_item.py` made in 4842a6d289873a4611151b787bf26518a8913589 explicitly tests for the bug reported in #10 by reading in the CIF
file and testing for the problematic string. Notice the commit was separate and included the failing test message,

Wherever possible in my work I follow Test Driven Development. So when dealing with a bug:

* I track down the bug.
* Write a test that fails because of the bug
*  Commit the test and make sure the project's CI/CD testing pipeline fails. I guess that as this project has .gitlab-ci.yml https://github.com/PDBeurope/ccdutils/blob/master/.gitlab-ci.yml there is a pipeline running in a separate internal GitLab repo?

**Comment by osmart** [comment]
Looking at your changes in detail:

the test changes made in https://github.com/PDBeurope/ccdutils/pull/12/commits/ec0c75c73abf168aaf67822fa7855a23328b67e2
pass presumably because the fix 706224b68e8afab0c089f2f7bb846b9efd20b87a was committed before the test change? This seems the wrong way around to me.

**Comment by osmart** [comment]
Dear Roshan,

I hope you find the comments made above to be useful. They are not intended to be patronizing in any way!

Best regards,

Oliver

**Comment by osmart** [comment]
> ## Keep commits atomic and make the 1st line of your commit messages human-readable.
> Change one thing at a time. Your message for this commit starts "Pdbecif #12" this is meaningless to humans - the commit message says what the commit actually does in human terms. It is great to include the issue number as well perhaps further down. You commit message includes a list of 5 things that are done. I would suggest it is better to do 5 commits.
> 
> There are a number of reasons why this is a good idea:
> 
> * if you ever need to revert a change (either permanently or for a test) this can be easily achieved. If you commit mulitple things in the same commit then this becomes difficult.
> * git blame is much more useful
> 
> Remember that code is "code is written once but read many times". Using git commits carefully helps this process.

Apologies I had not picked up that 6fa779f8d6564a65fa41471168ef45570a3dff2b was a pull request or merge commit of separate individual commits. My fault.

Maybe if you are doing open-source code development then explaining what you do in an issue would be useful.

**Pull request merged**